tests/xmlrs_reader_tests.rs

```diff
@@ -403,21 +404,21 @@ fn namespace_name(n: ResolveResult, name: QName, decoder: Decoder) -> String {
     let name = decoder.decode(name.as_ref()).unwrap();
     match n {
         // Produces string '{namespace}prefixed_name'
-        ResolveResult::Bound(n) => format!("{{{}}}{}", from_utf8(n.as_ref()).unwrap(), name),
+        ResolveResult::Bound(n) => format!("{{{}}}{}", decoder.decode(n.as_ref()).unwrap(), name),
         _ => name.to_string(),
     }
 }
 
-fn make_attrs(e: &BytesStart) -> ::std::result::Result<String, String> {
+fn make_attrs(e: &BytesStart, decoder: Decoder) -> ::std::result::Result<String, String> {
     let mut atts = Vec::new();
     for a in e.attributes() {
         match a {
             Ok(a) => {
                 if a.key.as_namespace_binding().is_none() {
                     atts.push(format!(
                         "{}=\"{}\"",
-                        from_utf8(a.key.as_ref()).unwrap(),
-                        from_utf8(&*a.unescaped_value().unwrap()).unwrap()
+                        decoder.decode(a.key.as_ref()).unwrap(),
+                        decoder.decode(&*a.unescaped_value().unwrap()).unwrap()
```

Comment:
This seems like it would be another instance of the incorrect "unescape, then decode" behavior mentioned, yes?

Reply:
Yeah, you right. Need to fix that too